recipe/2024-01-global-scope-secret-store.md

```diff
@@ -0,0 +1,202 @@
+# Adding support to extend the capabilities of secretStores to a global scope.
+
+* **Status**: Pending/Approved
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today we have Application.Core/secretStores resource to securely manage secrets for the Application. However a gap exists if user wants to create a secret before application or environment is created. To address this limitation, we need to enhance the capabilities of the existing secretStores resource. The objective is to extend its support to a global scope, enabling users to store secrets prior to the creation of the application or environment.
```

Comment:
By asking the users to provide `resource` property in the format of <namespace>/<secretName> if secret stores is a global scoped.

https://github.com/radius-project/design-notes/pull/38/files#diff-89d84238f05b572cd71ded0059391247faab67f69d3d9232b31350325b946cfaR125:~:text=%2B%20%20%20resource%3A%20%3Cnamespace%3E/%3C-,secretName,-%3E

Reply:
I'm not sure if we're talking about the same thing. Let me try being more specific.

If I create an `Applications.Core/secretStores` resource today and I **don't** specify the kubernetes secret name + namespace, then it will use the namespace of the application (or environment).

For the global secret case, there's no application or environment, so there's no way to look up the namespace we're going to use.

Reply:
From reading more of the doc, it looks like you did try to address this feedback as part of the proposal. Let's talk about it 👍